### PR TITLE
MM-61968 - a11y rhs focus management

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds the logic for the RHS component to get the focus during the mount and return back to the previous active element on unmount. This flow is working completely fine when navigating with the tabs keys and pressing enter for:
- pinned post
- recent mentions
- saved messages
- channel info
- members
- channel files

*known problem: when using tab to open search modal and performing search, the focus correctly gets set to the RHS panel, but if tabbing you close the RHS panel, the focus do not get back to the top center search button. This is already registered and tracked via: https://mattermost.atlassian.net/browse/MM-62226

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-62226

#### Screenshots

https://github.com/user-attachments/assets/4c724cd4-b9fd-4254-8b81-95a4520a1443

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by hmhealey — CHANGES_REQUESTED** [review]
This looks great! I like the idea of just focusing the container for the RHS in this case since it'll cover anything that gets put in the RHS. I didn't realize that `.focus()` could focus an element with `tabindex="-1"`, so that's a really clever way of doing making this work without messing with the regular focus flow

In addition to my comments below, could you add some E2E tests for this? We don't need to cover every possible RHS, but I think a few of them would be good so that we'll know if anything major breaks with them

File: webapp/channels/src/components/sidebar_right/sidebar_right.tsx
Comment (pvev) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:145]: yeah, long validation, will see if can place it in a external function or something

File: webapp/channels/src/components/sidebar_right/sidebar_right.tsx
Comment (pvev) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:172]: well, you are right that the RHS allows you to move around even when opened, but I still find valuable the option that when closing you get back to the previous point where you were. I was playing with the focus on other places, and still find valuable to be able to move (while RHS open) and when deciding to close the RHS via focus, getting back to the place that initially triggered the open.
And in the case that you choose other element that triggers the opening of the RHS, then the last element would be that one and the focus will be back to it (I tried opening recent mentions, then moving around until channel members, then close the RHS and the focus got back to channel members button). 
And lastly, when switching teams, RHS is closed and the focus is restarted to the opened channel post box, so no problem in that scenario then.

File: webapp/channels/src/components/sidebar_right/sidebar_right.tsx
Comment (pvev) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:144]: right now the plugin buttons do not open RHS when focus the plugin button + pressing enter. This needs to be tracked in a different ticket. I didn't find one specifically for this scenario but there are too many tickets, maybe @marianunez remembers if there is already one for this or should I create one.

Comment (pvev) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:144]: but yes, once this is working, changing the pluginId should also be considered in the scenario for contentChanged.

Comment (marianunez) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:144]: > right now the plugin buttons do not open RHS when focus the plugin button + pressing enter. 

I don't believe we have a ticket for this, feel free to create one. 

> Does this maybe need a case for when the RHS plugin changes? 

However, I believe @hmhealey was referring to when the RHS is open in pluggable mode (not necessarily from the plugin buttons of the apps bar) if the focus set properly

**Review by larkox — APPROVED** [review]

**Comment by pvev** [comment]
@hmhealey @marianunez These are the tickets for tracking the remaining work:
- Add E2E tests for RHS Focus via Tab: https://mattermost.atlassian.net/browse/MM-62288
- Plugin buttons not opening RHS on focus + Enter : https://mattermost.atlassian.net/browse/MM-62289

Proceeding to merge this then to unblock partners. Thanks!

**Comment by pvev** [comment]
somehow this change was affecting one e2e that is skipped in this commit: https://github.com/mattermost/mattermost/pull/29591/commits/09eeeb772aec7a71f6fcd1c9214c43ad8a916658

I will fix this as part of https://mattermost.atlassian.net/browse/MM-62288 

@hmhealey can I get an approval here?

File: webapp/channels/src/components/sidebar_right/sidebar_right.tsx
Comment (hmhealey) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:172]: We discussed more of this in a DM. I'm still thinking not sure about refocusing the previous element when the RHS is closed, but since LevelAccess is treating the RHS as a dialog, this behaviour is needed for that

**Review by hmhealey — APPROVED** [review]

File: webapp/channels/src/components/sidebar_right/sidebar_right.tsx
Comment (hmhealey) [inline webapp/channels/src/components/sidebar_right/sidebar_right.tsx:74]: I don't think it's likely a big deal, but I did want to mention that this could hold onto a reference to a DOM node after it gets detached from the DOM. Given we only hold onto a single node at once, I doubt this would cause us to leak memory, but I wanted to mention it in case it does come up later. If that ever does cause an issue, we could probably fix that by using a WeakRef anyway

Comment (pvev) [reply webapp/channels/src/components/sidebar_right/sidebar_right.tsx:74]: noted 🖊️

**Pull request merged**
